rpc/eth_api.go

```diff
@@ -555,22 +563,45 @@ func (e *PublicEthAPI) doCall(
 		data = []byte(*args.Data)
 	}
 
-	// Set destination address for call
-	var toAddr sdk.AccAddress
-	if args.To != nil {
-		toAddr = sdk.AccAddress(args.To.Bytes())
+	// Get nonce (sequence) from account
+	from := sdk.AccAddress(addr.Bytes())
+	accRet := authtypes.NewAccountRetriever(e.cliCtx)
+
+	if err := accRet.EnsureExists(from); err != nil {
+		// account doesn't exist
+		return nil, fmt.Errorf("nonexistent account %s: %w", addr.String(), err)
+	}
```

Comment:
I can no longer do `eth_call` from any account, which is an issue

eg. on development from an arbitrary address:
```
$ geth attach http://localhost:8545
> eth.call({to: "0xFB60B466315d85371f7b37064A9BFf088528cC74", from: "0x1d91a826aa03326ba13b8c954bfef8cfc7ba7a86"})
"0x"
```

now:
```
> eth.call({to: "0xFB60B466315d85371f7b37064A9BFf088528cC74", from: "0x1d91a826aa03326ba13b8c954bfef8cfc7ba7a86"})
Error: nonexistent account 0x1D91a826Aa03326Ba13B8C954BfEF8cFc7ba7a86: unknown address: account eth1rkg6sf42qvexhgfm3j25hlhcelrm575xhfhsm5 does not exist
	at web3.js:6347:37(47)
	at web3.js:5081:62(37)
	at <eval>:1:9(8)
```

this may cause some tooling issues, since it's expected that a call can happen from any account